src/DynamoCore/Migration/Migration.cs

```diff
@@ -251,9 +251,18 @@ orderby result.From
                 string typeName = elNode.Attributes["type"].Value;
                 typeName = Nodes.Utilities.PreprocessTypeName(typeName);
 
+                string nickName = elNode.Attributes["nickname"].Value;
+                typeName = Nodes.Utilities.PreprocessTypeName(typeName);
+
+                // If we try to migrate Function node, then use its' nickname as type.
+                if (typeName == "Dynamo.Nodes.DSFunction" && nodeMigrationLookup.ContainsKey(nickName))
```

Comment:
hmm. Does this have any effect on loading migrations for zero touch nodes? As they are also defined as Dynamo.Nodes.DSFunction?